InMobi/src/main/java/com/mopub/mobileads/InMobiAdapterConfiguration.kt

```diff
@@ -0,0 +1,243 @@
+                            placementIdErrorMessage)
+                }
+                return placementIdLong
+
+            } catch (e: NumberFormatException) {
+                throw Exception("InMobi Placement ID parameter is incorrect, cannot cast it to Long, it has to be a long value. " +
+                        placementIdErrorMessage)
+            }
+        }
+
+        /**
+         * Call for a load or interaction related failure an error.
+         *
+         * @param loadListener - Populate only if the failure is load related.
+         * @param interactionListener - Populate only if the failure is interaction related.
+         */
+        fun onInMobiAdFailWithError(error: Exception,
+                                    moPubErrorCode: MoPubErrorCode,
+                                    errorMessage: String?,
+                                    adapterName: String,
+                                    loadListener: AdLifecycleListener.LoadListener?,
+                                    interactionListener: AdLifecycleListener.InteractionListener?) {
+            MoPubLog.log(AdapterLogEvent.CUSTOM_WITH_THROWABLE, error)
+            onInMobiAdFail(errorMessage, moPubErrorCode, adapterName, loadListener, interactionListener)
+        }
+
+        /**
+         * Call for a load or interaction related failure with an event.
+         *
+         * @param loadListener - Populate only if the failure is load related.
+         * @param interactionListener - Populate only if the failure is interaction related.
+         */
+        fun onInMobiAdFailWithEvent(logEvent: AdapterLogEvent,
+                                    placementId: String,
+                                    moPubErrorCode: MoPubErrorCode,
+                                    errorMessage: String?,
+                                    adapterName: String,
+                                    loadListener: AdLifecycleListener.LoadListener?,
+                                    interactionListener: AdLifecycleListener.InteractionListener?) {
+            MoPubLog.log(placementId, logEvent, adapterName, moPubErrorCode.intCode, moPubErrorCode)
+            onInMobiAdFail(errorMessage, moPubErrorCode, adapterName, loadListener, interactionListener)
+        }
+
+        private fun onInMobiAdFail(errorMessage: String?,
+                                   moPubErrorCode: MoPubErrorCode,
+                                   adapterName: String,
+                                   loadListener: AdLifecycleListener.LoadListener?,
+                                   interactionListener: AdLifecycleListener.InteractionListener?) {
+            errorMessage?.let {
+                MoPubLog.log(CUSTOM, adapterName, it)
+            }
+
+            loadListener?.onAdLoadFailed(moPubErrorCode)
+            interactionListener?.onAdFailed(moPubErrorCode)
+
+        }
```

Comment:
It feels like lot of things are merged into `InMobiAdapterConfiguration` file which is not meant to be present in this class, as the responsibility of this class is not supposed to handle these utility methods and that too specific to ad load.

Reply:
They have done this for other partners as well. If you check AdMob's code they are building an Ad request builder in the adapter configuration class with "npa"